library/psa_crypto_cipher.c

```diff
@@ -263,7 +263,7 @@ const mbedtls_cipher_info_t *mbedtls_cipher_info_from_psa(
 {
     mbedtls_cipher_mode_t mode;
     psa_status_t status;
-    mbedtls_cipher_id_t cipher_id_tmp;
+    mbedtls_cipher_id_t cipher_id_tmp = MBEDTLS_CIPHER_ID_NONE;
```

Comment:
I and our CI can't reproduce the warning with `gcc-12 -Wall -Wextra` (which includes `-Wmaybe-uninitialized`). How did you get it? But explicitly initializing a variable isn't a bad thing anyway.

This code also exists in the `mbedtls-3.6` and `mbedtls-2.28` branches, so please make backport pull requests to these branches. By the way, please keep the “PR checklist” section in the pull request template, we need it for gatekeeping.

Reply:
I can't reproduce this with `gcc 12.0.1`, so I presume it's new with `gcc 12.2`

Reply:
On my machine and the CI, it's `gcc-12 (Ubuntu 12.3.0-1ubuntu1~22.04) 12.3.0`